### PR TITLE
Binding Default(null) value implemented (the value that shows when BindingC…

### PR DESCRIPTION
**Pull request by akgulebubekir**
### Description of Change

`NullValue`is a `System.Windows.Forms.Binding` class property that: holds and shows a value that if property is null or nor setted.
### Bugs Fixed
- 
### API Changes

Binding class have NullValue property now.

Added:
- object BindingBase.NullValue{get;set;} // property

Changed:
- 
### Behavioral Changes

If BindingContext is null NullValue binds instead of data.

### CODE REVIEW

**Comment by andreinitescu** [comment]
This is useful.
I assume this is the equivalent of `TargetNullValue` on Windows? 
Can I suggest to name it `TargetNullValue` like on Windows?

**Comment by andreinitescu** [comment]
This isn't `Fallbackvalue` like on Windows right?

**Comment by adrianknight89** [comment]
You should fix the indentation errors. First, start keeping tabs. `Tools -> Options -> Text Editor -> C# -> Tabs -> Keep tabs`. Then, you could fix the styling.

File: Xamarin.Forms.Core/Binding.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public Binding()
 		{
 		}
 
-		public Binding(string path, BindingMode mode = BindingMode.Default, IValueConverter converter = null, object converterParameter = null, string stringFormat = null, object source = null)
+		public Binding(string path, BindingMode mode = BindingMode.Default, IValueConverter converter = null, object converterParameter = null, string stringFormat = null, object source = null, object nullValue = null)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Binding.cs]: this is a breaking change. if you want this, you'll have to add a new `ctor`

File: Xamarin.Forms.Core/Binding.cs [hunk 2]
```diff
@@ -113,9 +116,12 @@ internal override void Apply(bool fromTarget)
 		internal override void Apply(object newContext, BindableObject bindObj, BindableProperty targetProperty)
 		{
 			object src = _source;
-			base.Apply(src ?? newContext, bindObj, targetProperty);
+			if (newContext == null && src == null)
+				base.Apply(NullValue, bindObj, targetProperty);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Binding.cs]: could be expressed as `src ?? newContext ?? NullValue`

File: Xamarin.Forms.Core/Binding.cs [hunk 3]
```diff
@@ -36,6 +36,7 @@ public Binding(string path, BindingMode mode = BindingMode.Default, IValueConver
 			Mode = mode;
 			StringFormat = stringFormat;
 			Source = source;
+			NullValue = nullValue;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Binding.cs]: should be done in base ctor, as you defined `NullValue` as a property of `BindingBase`

File: Xamarin.Forms.Core/Binding.cs (hunk 2 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Binding.cs]: if `NulValue` belongs to `BindingBase`, you shouldn't have to handle it in `Binding` at all

File: Xamarin.Forms.Core/Binding.cs (hunk 1 above)
Comment (djcparker) [inline Xamarin.Forms.Core/Binding.cs]: out of interest, why is it a breaking change given it is an optional parameter?

**Comment by jassmith** [comment]
I am completely sold on the need of this feature. Can you explain the use case in better detail with some examples so we can better consider its inclusion?

**Comment by andreinitescu** [comment]
@jassmith it's a useful and common feature present in all Windows platforms WPF\SL\WinPhone\Win8\UWP

**TargetNullValue**
https://msdn.microsoft.com/en-us/library/system.windows.data.bindingbase.targetnullvalue(v=vs.110).aspx
When a source object's property is null, the TargetNullValue is the alternate value you want to set for the target. So, whatever value is set for TargetNullValue will be set for the target. 

`<TextBlock Text="{Binding Name, TargetNullValue='No user selected'}" />`

**FallbackValue**
https://msdn.microsoft.com/en-us/library/system.windows.data.bindingbase.fallbackvalue%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396
This is used when a binding cannot determine a value at all, based on the the data source and the path. Or in other words, FallbackValue is used when the property bound to the source is not at all available. In that case, the value supplied for FallbackValue will be considered at the target end.

`<TextBlock Text="{Binding User.Name, TargetNullValue='No user selected'}" />`

Currently in Xamarin Forms, without these, you need to add not so nice workarounds which makes implementation bloated.

**Comment by andreinitescu** [comment]
I'm not sure though if the OP of the PR really did this properly

**Comment by StephaneDelcroix** [comment]
@andreinitescu my main problem with the current implementation is that's a mixed bag of TargeNullValue, FallBackValue and some other stuff, and that'll produce unpredictable results

**Comment by andreinitescu** [comment]
@StephaneDelcroix agreed, I saw that too. I was just responding to @jassmith question how is this useful

**Pull request closed without merging**